packages/@aws-cdk/pipelines/test/compliance/assets.test.ts

```diff
@@ -743,27 +743,27 @@ describe('pipeline with single asset publisher', () => {
 
     function THEN_codePipelineExpectation() {
       // THEN
-      const buildSpecName = Capture.aString();
-      expect(pipelineStack).toHaveResourceLike('AWS::CodePipeline::Pipeline', {
-        Stages: arrayWith({
+      const buildSpecName = new Capture();
+      Template.fromStack(pipelineStack).hasResourceProperties('AWS::CodePipeline::Pipeline', {
+        Stages: Match.arrayWith([{
           Name: 'Assets',
           Actions: [
             // Only one file asset action
-            objectLike({ RunOrder: 1, Name: 'FileAsset' }),
+            Match.objectLike({ RunOrder: 1, Name: 'FileAsset' }),
           ],
-        }),
+        }]),
       });
-      expect(pipelineStack).toHaveResourceLike('AWS::CodeBuild::Project', {
+      Template.fromStack(pipelineStack).hasResourceProperties('AWS::CodeBuild::Project', {
         Environment: {
           Image: 'aws/codebuild/standard:5.0',
         },
         Source: {
-          BuildSpec: buildSpecName.capture(stringLike('buildspec-*.yaml')),
```

Comment:
is this an assertion? does it need to be reproduced below? (similar with other captures in other tests)

Reply:
Maybe. It feels like the more important thing is the contents of the buildspec (per assertions below) over the name of the file, no? The name can change to be anything.

I can add it if you feel strongly.

Reply:
I have no idea about the original purpose of the test. Does it matter that the build spec has a particular name? Given that we have an assertion saying it does, it's probably important for some reason. Maybe @rix0rrr has an idea?

Reply:
I don't mind adding another assertion, and I've done so, however, it's still not clear to me why.